src/librbd/AioObjectRequest.h

```diff
@@ -93,6 +96,7 @@ class AioObjectRequest : public AioObjectRequestHandle {
   Context *m_completion;
   Extents m_parent_extents;
   bool m_hide_enoent;
+  const blkin_trace_info *m_trace_info;
```

Comment:
looking at your `aio_write_test_data_traced()` test as an example, it's starting the rbd write and waiting for the completion at the same time:
```c++
void aio_write_test_data_traced(...) {
  struct blkin_trace_info trace_info;
  rbd_aio_write_traced(image, off, len, test_data, comp, &trace_info);
  rbd_aio_wait_for_complete(comp);
```
we're passing a pointer to a `blkin_trace_info` that's on the stack here. but it's not a problem, because the stack stays valid until the operation completes - this operation is basically synchronous. but consider an asynchronous case, where we start the operation in one function, and wait for it later in another:
```c++
void start_write(...) {
  struct blkin_trace_info trace_info;
  rbd_aio_write_traced(image, off, len, test_data, comp, &trace_info);
}
void wait_for_write(...) {
  rbd_aio_wait_for_complete(comp);
}
```
again we pass a pointer to `blkin_trace_info` on the stack, but this time it goes out of scope right after it starts the write. so if rbd holds on to this pointer and tries it dereference it later, it won't be pointing to valid memory. that's why we need to store a copy of the trace info instead of just the pointer